src/convection/incflo_compute_advection_term.cpp

```diff
@@ -109,13 +109,44 @@ incflo::compute_convective_term (Vector<MultiFab*> const& conv_u,
 
     // We now re-compute the velocity forcing terms including the pressure gradient,
     //    and compute the tracer forcing terms for the first time
-    if (m_advection_type != "MOL")
-    {
+    if (m_advection_type != "MOL") {
+        
         compute_vel_forces(vel_forces, vel, density, tracer, tracer);
 
-        if (m_godunov_include_diff_in_forcing)
-            for (int lev = 0; lev <= finest_level; ++lev)
-                MultiFab::Add(*vel_forces[lev], m_leveldata[lev]->divtau_o, 0, 0, AMREX_SPACEDIM, 0);
+        if (m_godunov_include_diff_in_forcing) {
```

Comment:
This should not be needed here either, since the advect_momentum flag is already taken into account in compute_divtau()

Reply:
Same logic as above -- while advect_momentum is taken care of in compute_divtau(), it is not in vel_forces computation ...

Reply:
This also looks fine - the forcing needs to be built as if it were 'velocity' because down in ComputFluxesOnBoxFromState, either force or rho*force (if advect_momentum=T) is passed in.